### ISSUE TITLE
Headless batch rollout of a trained controller → trace bundle for the frontend (format tbu-traces/1)

### ISSUE THREAD

**Issue by OriPekelman** [OriPekelman/toy#190]
Follows #189. Once a controller is trained we want to **roll it out many times, headless, and hand the traces to the frontend** to overlay — hundreds of starts, one picture. The frontend then never runs a forward pass; it draws what the engine drew, so there is no second controller implementation to keep in parity.

## The ask

A rollout command over a trained controller (weights + the #189 sidecar): pick a start scheme — the single point, the 15-start ensemble, or `N` seeded yard draws — roll each to termination under the run's plant conventions, and write **one trace bundle**. Every arm in #189 goes through the same command, so `ga` and `dfa_tb` traces are comparable by construction.

## The format — shared with the frontend, defined here once

```jsonc
{
  "format": "tbu-traces/1",
  "provenance": {
    "arm": "dfa_tb",                 // ga | bptt | frozen | dfa_tb | dfa_rx | emulator-bptt | jacobian-bptt
    "engine": "toy",                 // or "truck_backer_upper"
    "engine_git": "<sha>",
    "weights": "<path or sha256>",
    "train_seed": 7,
    "net": { "shape": [4, 9, 1], "activation": "logistic", "output_map": "2s-1", "obs": 4 },
    "objective": "d2_best_approach"  // or "demo_terminal", "sr_fitness_reconstructed"
  },
  "plant": { "ls": 14, "lc": 6, "u_max_deg": 70, "r": 3, "step_cap": 300,
             "dock_ref": "trailer", "wrap": "pi" },
  "columns": ["signal", "u", "x", "y", "tc", "ts", "clamped"],
  "runs": [
    {
      "id": 0,
      "start": { "x": 80.0, "y": 50.0, "ts": -1.5708, "tc": -1.5708, "scheme": "ensemble", "idx": 5 },
      "end": "docked",               // docked | wall | bound | cap
      "steps": 212,
      "best_d2": 3.1, "best_step": 210, "terminal_d2": 3.4,
      "clamp_count": 0, "path_len": 636.0,
      "trace": [ [0.12, 0.146, 79.1, 49.7, -1.55, -1.571, 0], ... ]   // one row per step, `columns` order
    }
  ]
}
```

Row 0 of `trace` is the state **after** the first step; `start` holds the state before it. Angles in radians, lengths in metres, `u` the physical steering angle, `signal` the controller output in [−1, 1]. Per-run summaries are carried **and** recomputable from the trace, so a reader can cross-check the engine's arithmetic against its own.

Sizing: 1000 yard runs × 300 steps × 7 values is ~2M numbers — 30–40 MB as JSON. Fine for a file, heavy for a browser tab. Two things make it workable without a second format: a `--stride k` that keeps every k-th row (plus the last and the best-approach row, always), and gzip on the way out. Don't invent a binary layout for this yet.

## Also emit the frontend's start grid

The yard draws must be **reproducible from the seed alone** — the frontend (`OriPekelman/truck_backer_upper#6`) and this repo's original-code CLI (`#5` there) will draw the same `N` starts from the same seed so bundles from different engines overlay on identical states. Use the plant's own `sample_yard!` RNG from #188 and record `start.seed` and `start.idx` per run.

Consumers: `OriPekelman/truck_backer_upper#6` (overlay), `#4` there (metrics table reads the per-run summaries).

**Comment by OriPekelman** [comment]
**Shipped in `e9323ae`.** `TRUCK_LOAD` + `TRUCK_TRACE` on the #189 runner, reachable as:

```
toy research train truck --load ctrl.json --trace out.json \
  --trace-scheme yard --trace-n 1000 --trace-seed 1234 --stride 10
```

`--load` skips training and rolls the loaded controller as-is (`updates: 0` in the provenance, so a trace bundle can never be mistaken for one from a run that also trained). `--trace` also works without `--load`, rolling the just-trained controller with no round trip.

**It shares `tk_rollout` with the training loop.** Your reason for the frontend consuming traces — no second controller implementation to keep in parity — applies just as much inside the engine, so the rollout calls the same function training calls rather than re-deriving a trajectory.

## Format

`tbu-traces/1` exactly as specified: `provenance` (arm, engine, engine_git, weights, train_seed, net{shape,activation,output_map,obs}, objective — plus `updates` and `stride`), `plant`, `columns`, `runs[]` with `start`/`end`/`steps`/`best_d2`/`best_step`/`terminal_d2`/`clamp_count`/`path_len`/`trace`. Row 0 is the state *after* the first step; angles in radians; `end` ∈ `docked|wall|bound|cap`, with `wall` distinguished from `bound` by testing all three rig points against the dock wall.

**The "recomputable from the trace" promise is gated, not just stated.** `gate-truck-lane` recomputes `best_d2`, `terminal_d2` and *every* `u` from the rows and fails on any mismatch.

## The two frontend-facing properties, both gated two-sided

- **Yard starts reproduce from the seed alone** (the plant's `sample_yard!`), so your bundle and `truck_backer_upper#5`'s will overlay on identical states. The gate also asserts a *different* seed moves them — otherwise "reproducible from the seed" is vacuous.
- **`--stride k` never drops the last row or the best-approach row.** Those are the two rows every summary is computed from; dropping them would leave a bundle whose own numbers can't be recovered from its own trace. Measured: k=10 keeps 32 rows of 300, 715 KB → 81 KB.

Also gated: the export→load round trip is **exact** (a reloaded controller reproduces its exporting run's eval bit-for-bit — a lossy load reads as a slightly worse arm, not as an error), and a 4-9-1 controller loaded into a 3-input net is **refused** naming both counts.

**No gzip in the runner** — it has no zlib, and I did not want to invent a binary layout you explicitly said not to invent yet. Pipe it: `gzip -9 bundle.json`.

Two bugs found writing this, both mine and both silent: `ARM_NAMES` was defined *below* the rollout block so it was nil there; and `rb.dump.sub("}", ...)` splices at the *first* closing brace — the nested `start` object's — so bundles parsed as far as run 0 and then died. The run object is now emitted field by field instead of reopened by string surgery.

---

## On #189's `terminal` decision — implemented, with the measurement it deserves

`TRUCK_LOSS` now defaults to `terminal`. Since the default now descends something other than the score, it is made loud rather than quiet: the provenance carries `loss=`, and every step line prints the descended `loss=` beside the scored `score_d2=`.

But I owe you what it actually buys, because I offered it to unblock the single point and **it does not do that**:

| | `best` | `terminal` |
|---|---|---|
| single point, d²<5 | 0/10 (plateau, no gradient) | **0/10 — still exactly 504.0** |
| `bptt` ensemble mean d² | **2.5** | 2798.7 |
| `bptt` ensemble dock5 | **0.911** | 0.311 |

`terminal` removes the plateau — the loss moves, `zero_grad` falls to 0 — but the evaluated single-point result is *still* exactly 504.0, 0/10 seeds, all four arms, at every LR in {3, 6, 12, 24} (optimum interior at 6.0, so not a grid-edge artifact). Only `ga` reaches d² < 5 there, 5/5 seeds.

So the cost is real (the ceiling arm loses two-thirds of its docking rate) and the intended benefit did not materialise. I've implemented your decision rather than re-litigating it — `--loss best` is one flag away and both are recorded in provenance, so nothing is lost either way. Worth a look before the arc's runs are committed to it.

Two gate defects the flip exposed and fixed: the `zero_grad` leg was asserting a property of `best` while running on the default (now pinned, and the other side asserted too), and the C-FIXTURE verdict declared "the fixture DISCRIMINATES" off a bare mean comparison — duly saying it for a **0.3% difference that bptt won on 1 of 3 paired seeds**. It now requires a majority of paired seeds *and* a >10% margin, and otherwise reports the difference as inside seed noise at the smoke budget.

Consumers `truck_backer_upper#6` and `#4` should have everything they need; tell me if the overlay wants a field the summaries don't carry.
